account_invoice_report_grouped_by_picking_hide_pickings/models/account_move.py

```diff
@@ -1,15 +1,56 @@
 # Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import models
+from odoo import api, models
 
 
 class AccountMove(models.Model):
     _inherit = "account.move"
 
+    @api.model
+    def _update_lines_grouped_by_picking(self, lines, removed):
+        """Updates the invoice lines after removing some of them
+        to keep the total quantity consistent,
+        by creating correction lines
+        or modifying old correction lines's quantity
+        """
+        # Store the removed qty
+        to_update = {}
+        for line in removed:
+            key = line["line"]
+            if not to_update.get(key):
+                to_update[key] = line["quantity"]
+            elif to_update[key] + line["quantity"] != 0:
```

Comment:
Is the amount an integer or a float? If it's a float, you'll need to perform the comparison using Odoo's built-in method.